monai/apps/nuclick/transforms.py

```diff
@@ -0,0 +1,469 @@
+        # Removing points out of image dimension (these points may have been clicked unwanted)
+        x_del_indices = {i for i in range(len(cx)) if cx[i] >= n or cx[i] < 0}
+        y_del_indices = {i for i in range(len(cy)) if cy[i] >= m or cy[i] < 0}
+        del_indices = list(x_del_indices.union(y_del_indices))
+        cx = np.delete(cx, del_indices)
+        cy = np.delete(cy, del_indices)
+
+        click_map[cy, cx] = 1
+        bounding_boxes = []
+        for i in range(len(cx)):
+            x_start = cx[i] - bb // 2
+            y_start = cy[i] - bb // 2
+            if x_start < 0:
+                x_start = 0
+            if y_start < 0:
+                y_start = 0
+            x_end = x_start + bb - 1
+            y_end = y_start + bb - 1
+            if x_end > n - 1:
+                x_end = n - 1
+                x_start = x_end - bb + 1
+            if y_end > m - 1:
+                y_end = m - 1
+                y_start = y_end - bb + 1
+            bounding_boxes.append([x_start, y_start, x_end, y_end])
+        return click_map, bounding_boxes
+
+    def get_patches_and_signals(self, img, click_map, bounding_boxes, cx, cy, m, n, bb=128):
+        # total = number of clicks
+        total = len(bounding_boxes)
+        img = np.array([img])  # img.shape=(1,3,m,n)
+        click_map = np.array([click_map])  # clickmap.shape=(1,m,n)
+        click_map = click_map[:, np.newaxis, ...]  # clickmap.shape=(1,1,m,n)
+
+        patches = np.ndarray((total, 3, bb, bb), dtype=np.uint8)
+        nuc_points = np.ndarray((total, 1, bb, bb), dtype=np.uint8)
+        other_points = np.ndarray((total, 1, bb, bb), dtype=np.uint8)
+
+        # Removing points out of image dimension (these points may have been clicked unwanted)
+        x_del_indices = {i for i in range(len(cx)) if cx[i] >= n or cx[i] < 0}
+        y_del_indices = {i for i in range(len(cy)) if cy[i] >= m or cy[i] < 0}
+        del_indices = list(x_del_indices.union(y_del_indices))
+        cx = np.delete(cx, del_indices)
+        cy = np.delete(cy, del_indices)
+
+        for i in range(len(bounding_boxes)):
+            bounding_box = bounding_boxes[i]
+            x_start = bounding_box[0]
+            y_start = bounding_box[1]
+            x_end = bounding_box[2]
+            y_end = bounding_box[3]
+
+            patches[i] = img[0, :, y_start : y_end + 1, x_start : x_end + 1]
+
+            this_click_map = np.zeros((1, 1, m, n), dtype=np.uint8)
+            this_click_map[0, 0, cy[i], cx[i]] = 1
+
+            others_click_map = np.uint8((click_map - this_click_map) > 0)
+
+            nuc_points[i] = this_click_map[0, :, y_start : y_end + 1, x_start : x_end + 1]
+            other_points[i] = others_click_map[0, :, y_start : y_end + 1, x_start : x_end + 1]
+
+        # patches: (total, 3, m, n)
+        # nuc_points: (total, 1, m, n)
+        # other_points: (total, 1, m, n)
+        return patches, nuc_points, other_points
+
+
+class PostFilterLabeld(MapTransform):
```

Comment:
Please add doc-string for the transform.

Thanks.

Reply:
Added